components/product/product-description.tsx

```diff
@@ -1,31 +1,51 @@
+'use client'
```

Comment:
I think changing whole `product-description` to client component has negative consequences.
Alternatively can we transfer state between `VariantSelector` and `AddToCart` via URL params and `useSearchParams`?

Reply:
Thank you for the heads-up. I've removed 'use client' in the product-description. Instead, I've introduced a useSelectedVariant hook which is now being used in both the Price and AddToCart components. I hope this aligns with the proposed approach.